tools/generate_code_cache.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+// Flags: --expose-internals
+
+// This file generates the code cache for builtin modules and
+// writes them into static char arrays of a C++ file that can be
+// compiled into the binary using the `--code-cache-path` option
+// of `configure`.
+
+const {
+  nativeModuleWrap,
+  builtinSource,
+  cannotUseCache
+} = require('internal/bootstrap/cache');
+
+const vm = require('vm');
+const fs = require('fs');
+
+const resultPath = process.argv[2];
+if (!resultPath) {
+  console.error(`Usage: ${process.argv[0]} ${process.argv[1]}` +
+                'path/to/node_code_cache.cc');
+  process.exit(1);
+}
+
+/**
+ * Format a number of a size in bytes into human-readable strings
+ * @param {number} num
+ * @return {string}
+ */
+function formatSize(num) {
+  if (num < 1024) {
+    return `${(num).toFixed(2)}B`;
+  } else if (num < 1024 ** 2) {
+    return `${(num / 1024).toFixed(2)}KB`;
+  } else if (num < 1024 ** 3) {
+    return `${(num / (1024 ** 2)).toFixed(2)}MB`;
+  } else {
+    return `${(num / (1024 ** 3)).toFixed(2)}GB`;
+  }
+}
+
+/**
+ * Generates the source code of definitions of the char arrays
+ * that contains the code cache and the source code of the
+ * initializers of the code cache.
+ *
+ * @param {string} key ID of the builtin module
+ * @param {Buffer} cache Code cache of the builtin module
+ * @return { definition: string, initializer: string }
+ */
+function getInitalizer(key, cache) {
+  const defName = key.replace(/\//g, '_').replace(/-/g, '_');
+  const definition = `static uint8_t ${defName}_raw[] = {\n` +
+                     `${cache.join(',')}\n};`;
+  const initializer = `
+  v8::Local<v8::ArrayBuffer> ${defName}_ab =
+    v8::ArrayBuffer::New(isolate, ${defName}_raw, ${cache.length});
```

Comment:
Not sure, but maybe as a future optimization we could use a single array buffer for all of these?

Reply:
@addaleax Do you mean creating one buffer ane using offsets on it when creating the Uint8Arrays? I think that's doable but I am not sure how big a performance impact that would be.

Reply:
Yes, that’s what I mean – I wouldn’t expect it to make a huge difference, and it’s definitely not something that needs to be thought about in this PR :)

Reply:
@joyeecheung For a reference, whenever it's tackled, you can check out [v8-compile-cache](https://github.com/zertosh/v8-compile-cache/blob/3700d11a6b311b77f8904ec55465eb002311bb04/v8-compile-cache.js#L111-L137) which uses a single buffer and an offset map.